api/v1beta1/keystoneendpoint.go

```diff
@@ -0,0 +1,200 @@
+	ctx context.Context,
+	h *helper.Helper,
+) (ctrl.Result, error) {
+	endpoint := &KeystoneEndpoint{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ke.endpoint.Name,
+			Namespace: ke.endpoint.Namespace,
+		},
+	}
+
+	// add finalizer
+	controllerutil.AddFinalizer(endpoint, h.GetFinalizer())
+
+	op, err := controllerutil.CreateOrPatch(ctx, h.GetClient(), endpoint, func() error {
+		endpoint.Spec = ke.endpoint.Spec
+		endpoint.Labels = util.MergeStringMaps(endpoint.Labels, ke.endpoint.Labels)
+
+		err := controllerutil.SetControllerReference(h.GetBeforeObject(), endpoint, h.GetScheme())
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			h.GetLogger().Info(fmt.Sprintf("KeystoneEndpoint %s not found, reconcile in %ds", endpoint.Name, ke.timeout))
+			return ctrl.Result{RequeueAfter: time.Duration(ke.timeout) * time.Second}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		h.GetLogger().Info(fmt.Sprintf("KeystoneEndpoint %s - %s", endpoint.Name, op))
+	}
+
+	// update the endpoint object of the KeystoneEndpoint type
+	ke.endpoint, err = GetKeystoneEndpointWithName(ctx, h, endpoint.GetName(), endpoint.GetNamespace())
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	ke.id = ke.endpoint.Status.EndpointIDs
+
+	return ctrl.Result{}, nil
+}
+
+// GetEndpointIDs - returns the openstack endpoint IDs
+func (ke *KeystoneEndpointHelper) GetEndpointIDs() map[string]string {
+	return ke.id
+}
+
+// GetConditions - returns the conditions of the keystone endpoint object
+func (ke *KeystoneEndpointHelper) GetConditions() *condition.Conditions {
+	return &ke.endpoint.Status.Conditions
+}
+
+// Delete - deletes a KeystoneEndpoint if it exists.
+func (ke *KeystoneEndpointHelper) Delete(
+	ctx context.Context,
+	h *helper.Helper,
+) error {
+
+	endpoint, err := GetKeystoneEndpointWithName(ctx, h, ke.endpoint.Name, ke.endpoint.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	err = h.GetClient().Delete(ctx, endpoint, &client.DeleteOptions{})
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		return err
+	}
+
+	// Endpoint is deleted so remove the finalizer.
+	controllerutil.RemoveFinalizer(endpoint, h.GetFinalizer())
+	if err := h.GetClient().Update(ctx, endpoint); err != nil && !k8s_errors.IsNotFound(err) {
+		return err
+	}
+
+	h.GetLogger().Info(fmt.Sprintf("KeystoneEndpoint %s in namespace %s deleted", endpoint.Name, endpoint.Namespace))
+
+	return nil
+}
+
+// GetKeystoneEndpointWithName func
+func GetKeystoneEndpointWithName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*KeystoneEndpoint, error) {
+
+	ke := &KeystoneEndpoint{}
+	err := h.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, ke)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ke, err
+		}
+		return ke, err
```

Comment:
There doesn't seem to be a difference between the `return` inside the `if` and the `return` outside of it

Reply:
right, not required. will remove the condition.

Reply:
done with commit `Simpify GetKeystone<*>WithName and DeleteKeystone<*>WithName`